### ISSUE TITLE
Deebot T9 seems not update area sensors

### ISSUE THREAD

**Issue by gpongelli** [DeebotUniverse/client.py#592]
### Checks

- [X] I have searched the existing issues and no issue is describing my issue
- [X] I have checked the [FAQ](https://deebot.readthedocs.io/home/faq/)
- [X] I have checked the [documentation](https://deebot.readthedocs.io/integrations/home-assistant)
- [X] I have installed the latest version

### The problem

Hi,
I've a Deebot T9 AIVI integrated with Home Assistant, but seems it does not update cleaned area and cleaned time spent in "real time" while cleaning is in progress.
Official Ecovacs iOS app shows that information while cleaning is in progress, so data should be available to be retrieved.

Is it possible to do so like official Ecovacs app ?

I've also post a feature request into [homeassistant community forum](https://community.home-assistant.io/t/deebot-integration-refresh-cleaned-area-while-cleaning/787660), but seems HA integration shows the content from events generated here.

if you need some trace/dump I could help but I need to be instructed to setup everything for this need.

Thanks.

### On which deebot device (vacuum) you have the issue?

Deebot T9 AIVI

### Which version of the deebot-client are you using?

8.4.0

### Country

it

### Continent

eu

**Comment by edenhaus** [comment]
Yes I need logs :) Otherwise I cannot see what the bot is sending

**Comment by gpongelli** [comment]
can you explain how logs can be gathered ? thanks.

**Comment by edenhaus** [comment]
Enable debug logs directly on the Ecovacs config entry and attach them here

**Comment by gpongelli** [comment]
Hi @edenhaus , sorry for this late reply , here it is the log file plus some screenshot
[home-assistant_ecovacs_2025-01-12T10-59-47.203Z.log](https://github.com/user-attachments/files/18389124/home-assistant_ecovacs_2025-01-12T10-59-47.203Z.log)
![IMG_2471](https://github.com/user-attachments/assets/24a1b19d-d87a-488a-a3f6-b0e339bb6abc)
![IMG_2472](https://github.com/user-attachments/assets/dc46b009-9981-46de-abcc-8b0120e80d83)
![IMG_2474](https://github.com/user-attachments/assets/d91803cb-5caf-4eec-b82f-3f9fbb1d49d8)

in the three images you can see that:

- first image -> cleaning is running at 11:48 CEST timezone (so 10:48 UTC), and it had already cleaned 6m^2 
- second image -> map sensor timestamp is nearly the time of previous screenshot , meanwhile "area_cleaned" reports 23m^2 (yesterday I've done some cleaning without debug mode)
- third image -> area_cleaned sensor last update was at 10:07 UTC , so before the actual cleaning was started and **it is not updated during active cleaning**, that's the main focus of this bug. Map instead is updated while cleaning, and this is the expected behavior as both those info are present into ecovacs app


ps 
also cleaning_duration sensor is not updated while cleaning, and it has as last timestamp the one at 10:07 of previous job, but this could be another bug if you want.


thank you.

**Comment by gpongelli** [comment]
Hi @edenhaus , have you investigated the log?

**Comment by gpongelli** [comment]
I'm trying to understand the log, I've found something interesting: my deebot is a T9 AIVI model, into the log I've found `'toType': '659yh8'` ; the file `hardware/deebot/659yh8.py` refers to `x5d34r.py`, and opening `x5d34r.py` the very first comment is` """DEEBOT OZMO T8 AIVI Capabilities."""`

I've looked for a `T9 Capabilities` comment, and I've found that it's present into `hardware/deebot/8kwdb4.py` , so who set the content into json file ?
Is it correct that `'toType': '659yh8'` maps to a T8 device instead of a T9 device (like mine) ?

my device type is also confirmed into the log by this row:
```
2025-01-12 11:07:18.907 DEBUG (Dummy-1) [deebot_client.authentication] Success calling api url=https://portal-eu.ecouser.net/api/appsvr/app.do, 
params=None, json={'userid': 'hlticoh62a4c8951', 'todo': 'GetGlobalDeviceList'}, response={'code': 0, 'todo': 'result', 'ret': 'ok', 'devices': [{'did': '[REMOVED]', 
'name': 'E07516176D09HPDD0355', 'class': '659yh8', 'resource': 'r7j2', 'company': 'eco-ng', 'bindTs': 1736595592831, 'service': {'jmq': 'jmq-ngiot-eu.dc.ww.ecouser.net', 
'mqs': 'api-ngiot.dc-eu.ww.ecouser.net'}, 'deviceName': 'DEEBOT T9 AIVI',  'icon': 'https://portal-ww.ecouser.net/api/pim/file/get/603f5147c8115100083364bc',
 'ota': True, 'UILogicId': 'T9_AF_G', 'materialNo': '110-2011-0101', 
'pid': '60361d6134bf63000719d2e1', 'product_category': 'DEEBOT', 'model': 'T9_AIVI_INT', 
```

so maybe also this change is needed for T9 devices, what do you think ?

**Comment by edenhaus** [comment]
Please be patient and wait. You don't need to ping me because I'm subscribed to each issue. I'm maintaining this integration/lib in my free time, and currently, I'm getting a new kitchen, and therefore I don't have a lot of free time

**Comment by gpongelli** [comment]
any news here ?

**Comment by gpongelli** [comment]
> I'm trying to understand the log, I've found something interesting: my deebot is a T9 AIVI model, into the log I've found `'toType': '659yh8'` ; the file `hardware/deebot/659yh8.py` refers to `x5d34r.py`, and opening `x5d34r.py` the very first comment is` """DEEBOT OZMO T8 AIVI Capabilities."""`
> 
> I've looked for a `T9 Capabilities` comment, and I've found that it's present into `hardware/deebot/8kwdb4.py` , so who set the content into json file ? Is it correct that `'toType': '659yh8'` maps to a T8 device instead of a T9 device (like mine) ?
> 

answering myself, the content of `hardware/deebot/8kwdb4.py` and  `hardware/deebot/x5d34r.py` are the same as of today.

it should be more logically correct to have `hardware/deebot/659yh8.py` pointing to `8kwdb4.py`, but from a software point of view there are no differences.

**Comment by gpongelli** [comment]
looking at this issue, I've found that `StatsEvent` handles the missing data.

from the log I've attached, there are only those lines:
```
2025-01-12 11:07:20.941 DEBUG (Dummy-1) [deebot_client.event_bus] Calling on_first_subscription callbacks for <class 'deebot_client.events.StatsEvent'>
...
2025-01-12 11:07:21.544 DEBUG (Dummy-1) [deebot_client.event_bus] Notify subscribers with StatsEvent(area=23, time=2061, type='auto')

```

it's the same as the screenshot from home assistant, but it's unaligned with ecovacs app.

why aren't there more of this event ? who fires this event ?

**Comment by edenhaus** [comment]
> why aren't there more of this event ? who fires this event ?

The bot is firing this event as with all messages. So the data comes directly from the bot

**Comment by gpongelli** [comment]
I've launched today another cleaning with debug enabled, here's other interesting things:

vacuum entity updates the time spent on cleaning in progress, meanwhile "sensor...._cleaning_duration" does not.

this screenshot show 41 and 42 cleaning minutes, and apps are almost aligned
![Image](https://github.com/user-attachments/assets/593e7f20-85bd-4868-a7ed-79fd5576791a)

meanwhile the sensor "cleaning_duration" is on a wrong value 
![Image](https://github.com/user-attachments/assets/a062dee5-b975-49c6-a0fd-6516f4ef57dc)


same error is for cleaned_area, here the details where the last update is during the last night

![Image](https://github.com/user-attachments/assets/d64ca215-b8b9-4c15-b3c3-ca105cee0bc6)

here it is the log file (zipped due to size)

[home-assistant_ecovacs_2025-04-23T11-13-10.025Z.log.zip](https://github.com/user-attachments/files/19869001/home-assistant_ecovacs_2025-04-23T11-13-10.025Z.log.zip)


into the log I've found something interesting, maybe they've changed how those data are handled: looking at the timestamp of the screenshot above, I've found the following log messages
```
2025-04-23 10:49:20.265 DEBUG (Dummy-1) [deebot_client.mqtt_client] Got message: topic=iot/atr/onStats/f266fb63-5747-4cca-b8fb-3c1ceda9bd3f/659yh8/r7j2/j, payload=b'{"header":{"pri":1,"tzm":480,"ts":"1745398159530","ver":"0.0.1","fwVer":"1.4.9","hwVer":"0.1.1"},"body":{"data":{"area":27,"time":2459,"cid":"105534963","start":"1745395672","type":"auto","enablePowerMop":1,"powerMopType":2,"aiopen":1,"aitypes":[9],"avoidCount":1}}}'
2025-04-23 10:49:20.265 DEBUG (Dummy-1) [deebot_client.event_bus] Event is the same! Skipping (AvailabilityEvent(available=True))
2025-04-23 10:49:20.265 DEBUG (Dummy-1) [deebot_client.device] Try to handle message onStats: b'{"header":{"pri":1,"tzm":480,"ts":"1745398159530","ver":"0.0.1","fwVer":"1.4.9","hwVer":"0.1.1"},"body":{"data":{"area":27,"time":2459,"cid":"105534963","start":"1745395672","type":"auto","enablePowerMop":1,"powerMopType":2,"aiopen":1,"aitypes":[9],"avoidCount":1}}}'
2025-04-23 10:49:20.265 DEBUG (Dummy-1) [deebot_client.messages] Unknown message "onStats"
```

as you can see there is the "area" (27 m^2) and the "time" (2459 sec /60=40,983 minutes) that are same information reported into iOS app; last but not least there's the warning about message onStat carrying those info that's not handled, so handling it can solve this issue.


let me know if I can help you also on development side.
